app/routes/goals_routes.py

```
@@ -0,0 +1,105 @@
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    #request_body = request.get_json()
+    request_body = validate_data(Goal, request.get_json())
+    response_body= {}
+
+    # if "title" not in request_body:
+    #    response_body = {"details": "Invalid data"}
+    #    return make_response(jsonify(response_body), 400)
+    # else:
+    new_goal = Goal(title=request_body["title"])
+        
+    db.session.add(new_goal)
+    db.session.commit()
+
+    response_body["goal"] = new_goal.to_dict()
+
+    return make_response(jsonify(response_body), 201)
+
+@goals_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals_response=[]
+    goals= Goal.query.all()
+        
+    goals_response = [goal.to_dict() for goal in goals]
+
+    return jsonify(goals_response)
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def read_one_goal(goal_id):
+    goal= validate_model(Goal,goal_id)
+    response_body = {}
+    
+    response_body["goal"] = goal.to_dict()
+
+    return response_body
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    goal= validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+    response_body = {}
+    goal.title = request_body["title"]
+
+    response_body["goal"] = goal.to_dict()
+
+    db.session.commit()
+
+    return jsonify(response_body)
+
+@goals_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'})
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST"])
+def post_tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    request_body= request.get_json()
+
+    task_ids= request_body["task_ids"]
+
+    #this way we check that the tasks with those ids actually exist in the db
+    tasks = Task.query.filter(Task.task_id.in_(task_ids)).all()
```

Nice idea to use the `in_` filter to get the relevant task instances. However, note that this would silently ignore invalid task ids included in the request. Maybe that's the desired behavior, but since we have a common behavior in other places where we look up the object related to an id to report an error if an invalid id is supplied, consider doing something along the lines of
```py
    tasks = [validate_model(Task, task_id) for task_id in task_ids]
```